// angular2-message-service/angular2-message-service.umd.js
(function (global, factory) {
	typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@angular/core'), require('@angular/common'), require('ionic-angular')) :
	typeof define === 'function' && define.amd ? define(['exports', '@angular/core', '@angular/common', 'ionic-angular'], factory) :
	(factory((global['angular2-message-service'] = {}),global.core,global.common,global.ionicAngular));
}(this, (function (exports,core,common,ionicAngular) { 'use strict';

var UiMessageService = (function () {
    /**
     * @param {?} injector
     */
    function UiMessageService(injector) {
        this.showingLoading = false;
        this.loadingCtrl = injector.get(ionicAngular.LoadingController);
        this.alertCtrl = injector.get(ionicAngular.AlertController);
    }
    /**
     * @param {?} message
     * @return {?}
     */
    UiMessageService.prototype.showLoading = function (message) {
        if (!this.showingLoading) {
            this.showingLoading = true;
            this.loader = this.loadingCtrl.create({
                content: message
            });
            this.loader.present();
        }
    };
    /**
     * @return {?}
     */
    UiMessageService.prototype.hideLoading = function () {
        if (this.showingLoading) {
            this.loader.dismiss();
            this.showingLoading = false;
        }
    };
    /**
     * @param {?} title
     * @param {?} subtitle
     * @return {?}
     */
    UiMessageService.prototype.showAlert = function (title, subtitle) {
        this.alert = this.alertCtrl.create({
            title: title,
            subTitle: subtitle,
            buttons: ['OK']
        });
        this.alert.present();
    };
    /**
     * @param {?} title
     * @param {?} message
     * @param {?} buttons
     * @return {?}
     */
    UiMessageService.prototype.showConfirm = function (title, message, buttons) {
        var /** @type {?} */ alert = this.alertCtrl.create({
            title: title,
            message: message,
            buttons: buttons
        });
        alert.present();
    };
    return UiMessageService;
}());
UiMessageService.decorators = [
    { type: core.Injectable },
];
/**
 * @nocollapse
 */
UiMessageService.ctorParameters = function () { return [
    { type: core.Injector, },
]; };
var MessageButton = (function () {
    function MessageButton() {
    }
    return MessageButton;
}());

var UiMessageServiceModule = (function () {
    function UiMessageServiceModule() {
    }
    /**
     * @return {?}
     */
    UiMessageServiceModule.forRoot = function () {
        return {
            ngModule: UiMessageServiceModule,
            providers: [
                {
                    provide: UiMessageService,
                    useFactory: function (injector) {
                        return new UiMessageService(injector);
                    },
                    deps: [core.Injector]
                }
            ]
        };
    };
    return UiMessageServiceModule;
}());
UiMessageServiceModule.decorators = [
    { type: core.NgModule, args: [{
                imports: [
                    common.CommonModule
                ],
                declarations: [],
                exports: []
            },] },
];
/**
 * @nocollapse
 */
UiMessageServiceModule.ctorParameters = function () { return []; };

exports.UiMessageServiceModule = UiMessageServiceModule;
exports.UiMessageService = UiMessageService;
exports.MessageButton = MessageButton;

Object.defineProperty(exports, '__esModule', { value: true });

})));
