// angular2-message-service/index.js
import { Injectable, Injector, NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { AlertController, LoadingController } from 'ionic-angular';

var UiMessageService = (function () {
    /**
     * @param {?} injector
     */
    function UiMessageService(injector) {
        this.showingLoading = false;
        this.loadingCtrl = injector.get(LoadingController);
        this.alertCtrl = injector.get(AlertController);
    }
    /**
     * @param {?} message
     * @return {?}
     */
    UiMessageService.prototype.showLoading = function (message) {
        if (!this.showingLoading) {
            this.showingLoading = true;
            this.loader = this.loadingCtrl.create({
                content: message
            });
            this.loader.present();
        }
    };
    /**
     * @return {?}
     */
    UiMessageService.prototype.hideLoading = function () {
        if (this.showingLoading) {
            this.loader.dismiss();
            this.showingLoading = false;
        }
    };
    /**
     * @param {?} title
     * @param {?} subtitle
     * @return {?}
     */
    UiMessageService.prototype.showAlert = function (title, subtitle) {
        this.alert = this.alertCtrl.create({
            title: title,
            subTitle: subtitle,
            buttons: ['OK']
        });
        this.alert.present();
    };
    /**
     * @param {?} title
     * @param {?} message
     * @param {?} buttons
     * @return {?}
     */
    UiMessageService.prototype.showConfirm = function (title, message, buttons) {
        var /** @type {?} */ alert = this.alertCtrl.create({
            title: title,
            message: message,
            buttons: buttons
        });
        alert.present();
    };
    return UiMessageService;
}());
UiMessageService.decorators = [
    { type: Injectable },
];
/**
 * @nocollapse
 */
UiMessageService.ctorParameters = function () { return [
    { type: Injector, },
]; };
var MessageButton = (function () {
    function MessageButton() {
    }
    return MessageButton;
}());

var UiMessageServiceModule = (function () {
    function UiMessageServiceModule() {
    }
    /**
     * @return {?}
     */
    UiMessageServiceModule.forRoot = function () {
        return {
            ngModule: UiMessageServiceModule,
            providers: [
                {
                    provide: UiMessageService,
                    useFactory: function (injector) {
                        return new UiMessageService(injector);
                    },
                    deps: [Injector]
                }
            ]
        };
    };
    return UiMessageServiceModule;
}());
UiMessageServiceModule.decorators = [
    { type: NgModule, args: [{
                imports: [
                    CommonModule
                ],
                declarations: [],
                exports: []
            },] },
];
/**
 * @nocollapse
 */
UiMessageServiceModule.ctorParameters = function () { return []; };

export { UiMessageServiceModule, UiMessageService, MessageButton };
